### ISSUE TITLE
ZeroDivisionError when the number of unmet alarms is reduced to 0

### ISSUE THREAD

**Issue by Yujie-Liu** [paulgazz/kmax#150]
example:
```
$ cd linux
$ git remote add rdma https://git.kernel.org/pub/scm/linux/kernel/git/rdma/rdma.git
$ git fetch rdma linus
$ git checkout 143a6252e1b8ab424b4b293512a97cca7295c182

$ kismet --selectees CONFIG_SND_SOC_RT1015P --selectors CONFIG_SND_SOC_MT8192_MT6359_RT1015_RT5682 -a arm64
INFO: Computing the build system id for the Linux source..
INFO: Build system id: 699b29d22dec
INFO: Kismet will analyze the select constructs of the architecture "arm64" for unmet direct dependency.
INFO: All times reported are measured using Python's time.perf_counter() utility.
INFO: Prefetching the architecture kclause formulas.
INFO: Architecture kclause formulas were loaded. (3.57sec)
INFO: Identifying the select constructs.
INFO: Identification of the select constructs was done in 0.03sec. 1 constructs were found. unmet safe due to syntactical pass: 0. alarms: 1.
INFO: Doing optimized SAT pass for 1 constructs
INFO: 19:51:30 Optimized SAT pass progress:  1/1 (%100).
INFO: Optimized SAT test was done in 0.01sec. 1 constructs with alarms were checked. unmet safe due to optimized SAT test: 1. alarms: 0.
INFO: Doing precise SAT pass for 0 constructs
Traceback (most recent call last):
  File "/env_kmax/bin/kismet", line 625, in <module>
    dump_precisetest_progress()
  File "/env_kmax/bin/kismet", line 619, in dump_precisetest_progress
    info("%s Precise SAT pass progress: %s." % (current_time, dumps_progress(checked, total_to_check)), ending="\r")
  File "/env_kmax/bin/kismet", line 379, in dumps_progress
    return '{done:{fill}{width}}/{total} (%{perc})'.format(done=done, fill=" ", width=len(str(total)), total=total, perc= int((done / total) *100))
ZeroDivisionError: division by zero
```

**Comment by paulgazz** [comment]
kismet has two quick checking phases before it does the full proof check.  It looks like kismet deemed the select construct safe in a quick checking phases, but isn't aware that it can just halt if there are no more constructs left.  It assumes that there will still be something to check after the quick checking phases.  Will double-check this is the case and fix soon.

**Comment by paulgazz** [comment]
Actually, the issue looks simpler: it's computing a percentage of the alarms in the full proof check phase, but there are 0 constructs to check, so it's a divide by zero.

**Comment by paulgazz** [comment]
Fixed by #152

**Comment by Yujie-Liu** [comment]
Thanks for the fix.
We have upgraded to v4.2 release and the issue is gone.
